controllers/selfnoderemediation_controller.go

```diff
@@ -243,6 +251,31 @@ func (r *SelfNodeRemediationReconciler) remediateWithResourceDeletion(snr *v1alp
 
 	r.logger.Info("TimeAssumedRebooted is old. The unhealthy node assumed to been rebooted", "node name", node.Name)
 
+	if waitTime, err := removeResource(node, snr); err != nil {
+		return ctrl.Result{}, err
+	} else if waitTime != 0 {
+		return ctrl.Result{RequeueAfter: waitTime}, nil
+	}
+
+	fencingCompleted := fencingCompletedPhase
+	snr.Status.Phase = &fencingCompleted
+	if err := r.Client.Status().Update(context.Background(), snr); err != nil {
+		if apiErrors.IsConflict(err) {
+			// conflicts are expected since all self node remediation deamonset pods are competing on the same requests
+			return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
+		}
+		r.logger.Error(err, "failed to mark SNR as fencing completed")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *SelfNodeRemediationReconciler) remediateWithResourceDeletion(snr *v1alpha1.SelfNodeRemediation) (ctrl.Result, error) {
+	return r.remediateWithResourceRemoval(snr, r.removeResourceByResourceDeletion)
```

Comment:
IIUC the method `remediateWithResourceRemoval` only accepts `snr` and returns  a `Duration` because it needs to fulfill the same method signature of `removeResourceByAddingOutOfServiceTaint`.
I think a nicer way to keep that method "cleaner" would be to wrap it with another method.
Either anonymous::
```
func (r *SelfNodeRemediationReconciler) remediateWithResourceDeletion(snr *v1alpha1.SelfNodeRemediation) (ctrl.Result, error) {
	return r.remediateWithResourceRemoval(snr, func(node *v1.Node, snr *v1alpha1.SelfNodeRemediation) (time.Duration, error) {
		return 0, r.removeResourceByResourceDeletion(node)
	})
}
```

Or explicit:
```
func (r *SelfNodeRemediationReconciler) remediateWithResourceDeletion(snr *v1alpha1.SelfNodeRemediation) (ctrl.Result, error) {
	return r.remediateWithResourceRemoval(snr, r.removeResourceByResourceDeletionWrapper)
}

//removeResourceByResourceDeletionWrapper snr is only used in order to match method signature required by remediateWithResourceRemoval
func (r *SelfNodeRemediationReconciler) removeResourceByResourceDeletionWrapper(node *v1.Node, snr *v1alpha1.SelfNodeRemediation) (time.Duration, error) {
	return 0, r.removeResourceByResourceDeletion(node)
}
```

Reply:
Thanks. I agree with you. I took defining a new method.